### PR TITLE
TST: Remove broken test for deprecated functionality

### PR DESCRIPTION
**Pull request by theOehrly**
Registering converters by default on import is deprecated since #28722.

The test did not fail as the subprocess call contains errors in the python code. Specifically, the code is in double and single quotes. This means that the python code passed to the subprocess call is never actually excecuted. Python in the subprocess is called to "excecute" a string. This is will simply finish with exit code 0 as there is nothing to do basically.
Additonally, the test code contains a stray closing bracket at the end which would make the code fail if it was excecuted. 

- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
can u fix the test instead

**Comment by theOehrly** [comment]
How would you want this test fixed? There is a test for registering the converters explicitly. I don't know what case still needs to be tested with this after converters are no longer automatically registered.

**Comment by jreback** [comment]
this is a test that is using a sub process so it's different - it makes sure that we are not corrupting things in import

**Comment by theOehrly** [comment]
As far as my understanding went, there is nothing converter related done on import anymore. That's why I thought the test is no longer needed. But you are suggesting that something about the imports still needs to have a test. 
I probably don't have a deep enough understanding of what is done there then. So somebody who knows this better should probably look at this to fix it properly.

File: pandas/tests/plotting/test_converter.py [hunk 1]
```diff
@@ -54,17 +54,6 @@ def test_timtetonum_accepts_unicode():
 
 
 class TestRegistration:
-    def test_register_by_default(self):
-        # Run in subprocess to ensure a clean state
-        code = (
-            "'import matplotlib.units; "
```
Comment (jreback) [inline pandas/tests/plotting/test_converter.py:60]: if you fix the quoting here, does this test pass?

Comment (theOehrly) [reply pandas/tests/plotting/test_converter.py:60]: No, the assertion will fail then. Converters for `pd.Timestamp` and others are not added to units anymore on import

Comment (jreback) [reply pandas/tests/plotting/test_converter.py:60]: ahh ok, then this is a truly test in need of removal (that was what i am confirming. thanks

Comment (jreback) [reply pandas/tests/plotting/test_converter.py:60]: though i think having this test that the converters are NOT registered would actually be fine here, e.g. reverse the test as this functionaility is removed. The reason we are doing this in a subprocess as its very easy to have the ordering of imports change whether the converters is registered or not.

so we can remove the test, or better yet, to fix it and assert that pd.Timestamp is NOT added would be great.

**Comment by jreback** [comment]
thanks @theOehrly

**Pull request merged**